src/Serilog.Sinks.Couchbase/Sinks/Couchbase/CouchbaseSink.cs

```diff
@@ -50,26 +58,55 @@ public class CouchbaseSink : PeriodicBatchingSink
         /// <param name="batchPostingLimit">The maximum number of events to post in a single batch.</param>
         /// <param name="period">The time to wait between checking for event batches.</param>
         /// <param name="formatProvider">Supplies culture-specific formatting information, or null.</param>
-        public CouchbaseSink(string[] couchbaseUriList, string bucketName, int batchPostingLimit, TimeSpan period, IFormatProvider formatProvider)
+        /// <param name="bucketPassword"></param>
+        public CouchbaseSink(List<Uri> couchbaseUriList, string bucketName, string bucketPassword, int batchPostingLimit, TimeSpan period, IFormatProvider formatProvider)
             : base(batchPostingLimit, period)
         {
-            if (couchbaseUriList == null) throw new ArgumentNullException("couchbaseUriList");
-            if (couchbaseUriList.Length == 0) throw new ArgumentException("couchbaseUriList");
-            if (couchbaseUriList[0] == null) throw new ArgumentNullException("couchbaseUriList");
+            if (couchbaseUriList == null) throw new ArgumentNullException(nameof(couchbaseUriList));
+            if (couchbaseUriList.Count == 0) throw new ArgumentException(nameof(couchbaseUriList));
+            if (couchbaseUriList[0] == null) throw new ArgumentNullException(nameof(couchbaseUriList));
+
+            if (bucketName == null) throw new ArgumentNullException(nameof(bucketName));
 
-            if (bucketName == null) throw new ArgumentNullException("bucketName");
+            var config = new ClientConfiguration();
+
+            // Clearing servers here because for some reason it's trying to auto-add localhost:8091/pools
+            config.Servers.Clear();
 
-            var config = new global::Couchbase.Configuration.CouchbaseClientConfiguration();
-            
             foreach (var uri in couchbaseUriList)
-                config.Urls.Add(new Uri(uri));
-            config.Bucket = bucketName;
+                config.Servers.Add(uri);
+
+            var cluster = new Cluster(config);
+
+            IBucket bucket = null;
+
+            try
+            {
+                // Not using BucketConfigs property off of the Cluster because the OpenBucket method is not
+                // accepting 0 arguments in 2.2.2.0 of the Couchbase SDK as it is documented that it should.
+                bucket = cluster.OpenBucket(bucketName, bucketPassword);
+            }
+            catch(CouchbaseBootstrapException exception)
+            {
+                throw new InvalidOperationException(exception.InnerException.Message);
+            }
+            finally
+            {
+                if (bucket != null)
+                {
+                    cluster.CloseBucket(bucket);
+                }
+                else
+                {
+                    throw new InvalidOperationException("bucket '" + bucketName + "' does not exist");
```

Comment:
This check can be made immediately after the assignment inside the `try` block. Putting it here means it will be executed when `OpenBucket` fails, and thus potentially throw and replace the original exception.